### PR TITLE
[CAT-81] support bytestream uploads

### PR DESCRIPTION
**Pull request by goatrocks**
Enable the ability to submit bytestreams for upload (and also check in retry improvements)

### CODE REVIEW

File: src/main/java/com/indico/mutation/WorkflowSubmission.java [hunk 1]
```diff
@@ -76,15 +87,30 @@ public WorkflowSubmission duplicationId(UUID id){
...
+                fileMetadata = this.upload(this.files);
+                for (Object f : fileMetadata) {
+                    JSONObject uploadMeta = (JSONObject) f;
+                    JSONObject meta = new JSONObject();
+                    meta.put("name", uploadMeta.getString("name"));
+                    meta.put("path", uploadMeta.getString("path"));
+                    meta.put("upload_type", uploadMeta.getString("upload_type"));
+                    FileInput input = FileInput.builder().filename(((JSONObject) f).getString("name")).filemeta(meta).build();
+                    files.add(input);
+            }
```
Comment (nkassis) [inline src/main/java/com/indico/mutation/WorkflowSubmission.java:101]: indentation

File: src/main/java/com/indico/mutation/WorkflowSubmission.java [hunk 2]
```diff
@@ -76,15 +87,30 @@ public WorkflowSubmission duplicationId(UUID id){
...
+            if(this.streams != null){
+                fileMetadata = this.uploadBytes(this.streams);
+                for (Object f : fileMetadata) {
+                    JSONObject uploadMeta = (JSONObject) f;
+                    JSONObject meta = new JSONObject();
+                    meta.put("name", uploadMeta.getString("name"));
+                    meta.put("path", uploadMeta.getString("path"));
+                    meta.put("upload_type", uploadMeta.getString("upload_type"));
+                    FileInput input = FileInput.builder().filename(((JSONObject) f).getString("name")).filemeta(meta).build();
```
Comment (nkassis) [inline src/main/java/com/indico/mutation/WorkflowSubmission.java:111]: do we allow them to set the filename from an arg?

**Review by nkassis — COMMENTED** [review]
Can we also add an example for this

File: src/main/java/com/indico/mutation/WorkflowSubmission.java (hunk 2 above)
Comment (goatrocks) [reply src/main/java/com/indico/mutation/WorkflowSubmission.java:111]: We do -- they send a map of string -> stream. key is used as a filename.

**Comment by goatrocks** [comment]
> Can we also add an example for this

Example added

**Pull request merged**